Added simple-auth to docker-compose

valentinyanakiev opened the pull request:
CT-Server is instantiated after Simple-Auth (it's a dependency).

valentinyanakiev committed:
Auto stash before merge of "develop" and "origin/develop"

Adding ct-auth-provider

valentinyanakiev committed:
Fixed docker-compose issues

valentinyanakiev committed:
Removed exposing MySQL from docker-compose

valentinyanakiev committed:
Minor fixes in simple-auth provider dependency

techsmyth commented on docker-compose.yaml:
```diff
@@ -72,6 +73,24 @@ services:
-
-
+  ct_auth_provider:
+    container_name: ct_simple_auth
```
should be ct_demo_simple_auth to match the names of other containers in this stack

valentinyanakiev replied:
Addressed in 015167d191a5b486281d58d9d73616ddcf90c8fe

techsmyth commented on docker-compose.yaml:
```diff
@@ -16,7 +16,7 @@ services:
       - DATABASE_HOST
       - MYSQL_ROOT_PASSWORD
-      - MYSQL_DATABASE
+      - MYSQL_DATABASE=cherrytwist
```
Presumably the simple auth config also needs to be passed in here?

valentinyanakiev replied:
It is passed, the context of the env variable is the container. Given the parameters are passed from .env file and the name of the env variable is the same, I am configuring it directly in docker-compose.

techsmyth commented on .env.default:
```diff
@@ -27,3 +27,7 @@ LOGGING_LEVEL=warn
 
 COMPOSE_PROJECT_NAME=cherrytwist-demo
+
+SERVICE_ENDPOINT_PORT=3002
```
Should these env vars not have names to show they are for the Simple Auth service?

valentinyanakiev replied:
Yes and no - they are configured in the context of their own service and shouldn't be aware of the larger ecosystem they will be part of. This should be addressed on a system level and merits a larger discussion (otherwise our env variables will be each 50+ characters...).

techsmyth reviewed (commented):
Looks like it all makes sense - a few suggested naming updates / env var passing comments...

valentinyanakiev committed:
Bootstrap with an admin role

Changed image dependency version for simple-auth to 0.1.5. Addressed minor PR comment.
